### PR TITLE
2021-07-08 GUARD-2014 GUARD-1846 NewEgg_InventorySync_FailedWith_CT055Error

### PR DESCRIPTION
**Pull request by maxim-saltanov-skuvault**
**Problem:**
Inventory Sync randomly fails with CT055 error.

I have reproduced the problem with the unit test (UpdateItemsInventory_WhenSkuAndQuantityDuplicatedAndDoesntExist_ShouldNotThrowCT055Error) and for the BusinessAccount only. As stated in the response error description https://developer.newegg.com/newegg_marketplace_api/item_management/update_inventory_and_price/ this happens if we duplicate update requests for a product that does not exist in NewEgg. 
image

**Solution**

I just added CT055 to the list of allowed errors, that is, the error will be logged in v1, but the integration will not break. Also I added some fixes across unit tests so that they are successfully passed at moment.

PS. I've merged the 1846 branch to the current (we have opened PR on it also and this ticket blocked them)

### CODE REVIEW

**Review by maxim-skuvault — COMMENTED** [review]
See inline comment

**Review by maxim-skuvault — COMMENTED** [review]
@maxim-saltanov-skuvault , 
1. Normally, Inventory Sync will verify that an item exists in the integration before pushing quantity. Are we not doing it for NewEgg? Any idea why we're pushing quantity to a non-existing product and also multiple times?
2. Since this branch includes 1846, can you add GUARD-1846 to the title of the PR, to make it more clear? From discussions, I didn't think we were going to release just GUARD-1846 without this fix
3. I also commented on the 2014 commit about "*_ShouldNotThrowCT055Error" tests needing an assert. Can't find that comment now 🤷

**Comment by maxim-saltanov-skuvault** [comment]
> @maxim-saltanov-skuvault ,
> 
> 1. Normally, Inventory Sync will verify that an item exists in the integration before pushing quantity. Are we not doing it for NewEgg? Any idea why we're pushing quantity to a non-existing product and also multiple times?

**I'm not sure why it works this way, but this is how it was implemented initially in the Regular NewEgg Account (and later was inherited for NewEgg Business too). I suppose we can create a separate ticket and investigate / improve this. What do you think?**

> 2. Since this branch includes 1846, can you add GUARD-1846 to the title of the PR, to make it more clear? From discussions, I didn't think we were going to release just GUARD-1846 without this fix

**Yes, sure, will do**

> 3. I also commented on the 2014 commit about "*_ShouldNotThrowCT055Error" tests needing an assert. Can't find that comment now 🤷

**I just created a PR today and couldn't delete anything for sure. Could you explain what do you mean please? We could wrap this unit test it in "Assert" if in some case we expected to see an error and consider the test successful in this case (that is, before I added the fix, we could do this) - what exactly do you propose to do for now when issue was fixed?**

**Comment by maxim-saltanov-skuvault** [comment]
> See inline comment

Hi Max, which comments should I check. I looked the code but can't see anyone.. may be am I missed something?

**Comment by maxim-skuvault** [comment]
@maxim-saltanov-skuvault , response to [this](https://github.com/skuvault-integrations/newEggAccess/pull/9#issuecomment-876675431) since can't reply in a thread :(
1. If this is how it works (pushes quantities without checking if items exist), then it's fine. No need to do anything. We must've done it this way for a reason, since it's outside the norm
2. Ok. Thanks
3. I just meant to add `Assert.DoesNotThrow(action)` or the FluentAssertions equivalent. To make it more explicit what we're asserting.

**Comment by maxim-saltanov-skuvault** [comment]
> @maxim-saltanov-skuvault , response to [this](https://github.com/skuvault-integrations/newEggAccess/pull/9#issuecomment-876675431) since can't reply in a thread :(
> 
> 1. If this is how it works (pushes quantities without checking if items exist), then it's fine. No need to do anything. We must've done it this way for a reason, since it's outside the norm

Yes, it is how it works (how it was implemented before)

> 2. Ok. Thanks

Done

> 3. I just meant to add `Assert.DoesNotThrow(action)` or the FluentAssertions equivalent. To make it more explicit what we're asserting.

Done

**Review by maxim-skuvault — APPROVED** [review]

**Review by jnewman-sv — APPROVED** [review]

File: src/NewEggAccess/NewEggAccess.csproj [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <PackageLicenseUrl></PackageLicenseUrl>
     <PackageProjectUrl>https://github.com/skuvault/newEggAccess</PackageProjectUrl>
     <RepositoryUrl>https://github.com/skuvault/newEggAccess</RepositoryUrl>
-    <Version>1.1.1.0-alpha</Version>
```
Comment (maxim-skuvault) [inline src/NewEggAccess/NewEggAccess.csproj]: @maxim-saltanov-skuvault , We normally don't check in -alpha into access. Only into the app, while it's being tested. We only publish access as -alpha from local

**Review by nathan-skuvault — APPROVED** [review]

**Review by chris-skuvault — APPROVED** [review]

**Pull request merged**